mico-core/src/main/java/io/github/ust/mico/core/web/ApplicationController.java

```diff
@@ -256,6 +239,96 @@
         return ResponseEntity.ok(new Resource<>(applicationDeploymentInformation));
     }
 
+    @GetMapping("/{" + PATH_VARIABLE_SHORT_NAME + "}")
+    public ResponseEntity<Resources<Resource<MicoApplication>>> getApplicationsByShortName(@PathVariable(PATH_VARIABLE_SHORT_NAME) String shortName) {
+        List<MicoApplication> micoApplicationList = applicationRepository.findByShortName(shortName);
+
+        List<Resource<MicoApplication>> applicationResourceList = getApplicationResourceList(micoApplicationList);
+
+        return ResponseEntity.ok(
+            new Resources<>(applicationResourceList,
+                linkTo(methodOn(ApplicationController.class).getApplicationsByShortName(shortName)).withSelfRel()));
+    }
+
+    @PatchMapping("/{" + PATH_VARIABLE_SHORT_NAME + "}/{" + PATH_VARIABLE_VERSION + "}/" + PATH_SERVICES)
+    public ResponseEntity<Void> addServiceToApplication(@PathVariable(PATH_VARIABLE_SHORT_NAME) String applicationShortName,
+                                                        @PathVariable(PATH_VARIABLE_VERSION) String applicationVersion,
+                                                        @RequestBody MicoService service) {
+        Optional<MicoApplication> applicationOptional = applicationRepository.findByShortNameAndVersion(applicationShortName, applicationVersion);
+        if (!applicationOptional.isPresent()) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND,
+                "Application '" + applicationShortName + "' '" + applicationVersion + "' does not exist!");
+        }
+        validateProvidedService(service);
+
+        MicoApplication application = applicationOptional.get();
+        if (!application.getServices().contains(service)) {
+            log.info("Add service '" + service.getShortName() + "' '" + service.getVersion() +
+                "' to application '" + applicationShortName + "' '" + applicationVersion + "'.");
+            application.getServices().add(service);
+            applicationRepository.save(application);
+        } else {
+            log.info("Application '" + applicationShortName + "' '" + applicationVersion +
+                "' already contains service '" + service.getShortName() + "' '" + service.getVersion() + "'.");
+        }
+        return ResponseEntity.noContent().build();
+    }
+
+    /**
+     * Returns a list of services associated with the mico application specified by the parameters.
+     *
+     * @param shortName the name of the application
+     * @param version   the version of the application
+     * @return the list of mico services that are associated with the application
+     */
+    @GetMapping("/{" + PATH_VARIABLE_SHORT_NAME + "}/{" + PATH_VARIABLE_VERSION + "}/" + PATH_SERVICES)
```

Comment:
When reordering endpoints put GET as the first endpoint of all endpoints that share the same path.